app/scripts/models/reliers/relier.js

```diff
@@ -4,25 +4,41 @@
 
 /**
  * A relier is a model that holds information about the RP.
+ *
+ * A subclass should override `fieldsInResumeToken` to add/modify which
+ * fields are saved to and populated from a resume token in the resume
+ * query parameter.
  */
 
 define([
-  'underscore',
+  'cocktail',
   'models/reliers/base',
+  'models/mixins/resume-token',
   'models/mixins/search-param',
   'lib/promise',
-  'lib/constants'
-], function (_, BaseRelier, SearchParamMixin, p, Constants) {
+  'lib/constants',
+  'underscore'
+], function (Cocktail, BaseRelier, ResumeTokenMixin, SearchParamMixin, p,
+  Constants, _) {
   'use strict';
 
+  var RELIER_FIELDS_IN_RESUME_TOKEN = ['campaign', 'entrypoint'];
+
   var Relier = BaseRelier.extend({
     defaults: {
-      service: null,
-      preVerifyToken: null,
-      email: null,
       allowCachedCredentials: true,
+      campaign: null,
+      email: null,
       entrypoint: null,
-      campaign: null
+      preVerifyToken: null,
+      service: null,
+      setting: null,
+      uid: null,
+      utm_campaign: null,
```

Comment:
Add these to the [query-params doc](https://github.com/mozilla/fxa-content-server/blob/master/docs/query-params.md).